### PR TITLE
Document tally recovery

### PR DESCRIPTION
**Pull request by mzabaluev**

### CODE REVIEW

File: doc/tally-recovery.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+to be rejected by the consensus leader nodes at block creation.
+
+To allow computation of the tally across all ballots legitimately submitted
+by the users in face of such discrepancies, an alternative
+tallying method was devised and implemented in the catalyst-toolbox tool.
+Afterwards, this method can be used to confirm the blockchain tally did not
+miss any received legitimate ballots that were recorded into persistent
+fragment logs.
+
+## Input: persistent fragment logs
```
Comment (eugene-babichenko) [inline doc/tally-recovery.md:16]: I think documenting the format or just adding the link to where it is defined would be good.

File: doc/tally-recovery.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+just like in the blockchain node (with some technical twists: a
+mirror set of accounts is created because we need some private keys to
+supply to the library code performing ledger state transitions).
+
+## Replay of transactions
+
+The tally recovery command parses the persistent fragment logs and applies
+the fragments in order they were received, to replay changes in the ledger state
+of a blockchain node that would process transactions to the same effect, if
+the account spending counters used to sign transactions spending form the same
```
Comment (Zeegomo) [inline doc/tally-recovery.md:56]: This last sentence is a bit confusing to me.

File: doc/tally-recovery.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+is low enough to ignore such occurrences.
+
+To simplify the logic, the recovery tool only processes fragments of
+these kinds:
+
+- Ballot (VoteCast) transactions with one account input and no outputs.
+  Private votes are not yet supported.
+- Plain value transfer transactions with one account input and one account
+  output. The recovery tool updates the voting power with the results of
+  the transfer and adds the output accounts.
```
Comment (Zeegomo) [inline doc/tally-recovery.md:82]: The tally tool also processes `VoteTally` transactions. To be precise, those are a bit different from other transactions in the sense that they are replayed without the use of mirror accounts.

File: doc/tally-recovery.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+Other kinds of fragments are not expected to be submitted by voting users,
+so the recovery tool only reports such fragments in warning messages.
+The party performing the tally recovery should to examine the atypical
+fragments to decide if they could affect the recovered tally.
+
+## Output: Tally
+
+After the fragment logs have been replayed, the vote plan status including
+the tally results is updated as it would be in the node's ledger state,
```
Comment (Zeegomo) [inline doc/tally-recovery.md:92]: The tally results are included only if there are tally transactions in the fragment logs

File: doc/tally-recovery.md (hunk 2 above)
Comment (mzabaluev) [reply doc/tally-recovery.md]: Tried to fix.

**Review by Zeegomo — APPROVED** [review]

**Pull request merged**